src/Core/tests/DeviceTests/Handlers/HandlerTestBaseOfT.Tests.cs

```diff
@@ -19,21 +20,35 @@ public abstract partial class HandlerTestBase<THandler, TStub>
 		[Fact(DisplayName = "Handlers Deallocate When No Longer Referenced")]
 		public async Task HandlersDeallocateWhenNoLongerReferenced()
 		{
-			// Once this includes all handlers we can't delete this
-			if (!typeof(TStub).IsAssignableTo(typeof(IEditor)))
+			// Once this includes all handlers we can delete this
+			Type[] testedTypes = new[]
+			{
+				typeof(EditorHandler),
+				typeof(DatePickerHandler)
+			};
+
+			if (!testedTypes.Any(t => t.IsAssignableTo(typeof(THandler))))
 				return;
 
-			var stub = new TStub();
-			WeakReference<TStub> weakView = new WeakReference<TStub>(stub);
+			var handler = await CreateHandlerAsync(new TStub()) as IPlatformViewHandler;
 
-			var handler = await CreateHandlerAsync(stub) as IPlatformViewHandler;
 			WeakReference<THandler> weakHandler = new WeakReference<THandler>((THandler)handler);
+			WeakReference<TStub> weakView = new WeakReference<TStub>((TStub)handler.VirtualView);
 
-			stub = null;
 			handler = null;
 
 			await AssertionExtensions.Wait(() =>
 			{
+#if !IOS
+				// When running on CI this seems to give the android GC the
```

Comment:
@jonpryor @jonathanpeppers  Do you have thoughts on a better way to write this test? I'm trying to verify that some classes aren't pinned by anything and will GC but I'm not having the best of luck making this test reliable. It works fine locally and on our API 30 CI tests but API 23 is flakey.   Is there a way to force the Android GC collect more aggressively? Or to validate that the classes I've instantiated won't cause a leak when all local references go out of scope? When I run this test locally on an API 23 device the tests all pass fine.

Reply:
This has an example the Mono team recommended: https://github.com/xamarin/java.interop/commit/220b87fbbb8d986c2c8951757aa3ea45d8e7e7eb

Reply:
AFAICT this isn't yielding different results for me. Here's what I know and what I've found

1) I can't use this out of the box because this runs the code on a `Thread` and some of our handlers need to be created on the UI Thread
2) For this particular handler if I comment out the code that creates the [dialog](https://github.com/dotnet/maui/blob/release/6.0.4xx-sr2/src/Core/src/Handlers/DatePicker/DatePickerHandler.Android.cs#L24) then this code is able to run on the thread and the tests pass on the CI.
3) This all passes fine locally if I leave that code in that creates the dialog so I don't want to just change this code if CI isn't giving me accurate data. It also passes fine on CI for API 30.
4) I tried to just use `async void` with `PerformNoPinAction` but I get a feeling from the comments that an important feature of this mechanism is that the stack is allocating to the thread itself so I think if I create the handler from the UIThread it sort of invalidates what's all going on here. 
5) Just an observation but I wonder if the devices on CI have trouble running the GC because the device being tested on is very limited and has a reduced number of threads available. In one variation of this test I was trying out Thread.Sleep() inside `PerformNoPinAction` just to give the UIThread time to create the handlers and then I could run `Collect` on the `Thread`. This worked fine locally but on CI the `Thread.Sleep` would cause the entire test run to just lockup until the `Thread.Sleep` finished.

Reply:
@PureWeen are you able to test locally with the exact same emulator CI is using? I think it is created with cake scripts, and you should be able to run those?

Reply:
Alright, I ended up just splitting the test up into two separates tests. One that allocates and then one that tests if those allocations are still present. That also fixes me on the WinUI side of things as well.

Reply:
It's still a wee bit quirky. If I only run the GC tests they all pass fine. If I add the rest of the tests back in it never looks like it has time to collect.

Reply:
> @PureWeen are you able to test locally with the exact same emulator CI is using? I think it is created with cake scripts, and you should be able to run those?

Alright, now that we have all the tools/etc. updated for windows :-)

I was able to get this to run and reproduce locally.
It's odd because it only recreates when running the tests via the cake script/xharness cli

```bash
dotnet cake eng/devices/android.cake --project "H:\github\maui2\src\Core\tests\DeviceTests\Core.DeviceTests.csproj" --device=android-emulator-32_23
```

If I use the visual runner it runs without any issues. Not sure yet what features of running via the CLI would cause this to be quirky.

Reply:
Because of what this is testing, I'm going to just skip API 23 for these tests. They pass on API 30 which proves the point.